nodeup/pkg/model/kube_proxy.go

```diff
@@ -120,30 +120,26 @@ func (b *KubeProxyBuilder) buildPod() (*v1.Pod, error) {
 		return nil, fmt.Errorf("Error parsing CPURequest=%q", c.CPURequest)
 	}
 
-	flags, err := flagbuilder.BuildFlags(c)
+	flags, err := flagbuilder.BuildFlagsList(c)
 	if err != nil {
 		return nil, fmt.Errorf("error building kubeproxy flags: %v", err)
 	}
-
 	image := c.Image
-	cmd := "echo -998 > /proc/$$$/oom_score_adj && kube-proxy --kubeconfig=/var/lib/kube-proxy/kubeconfig --conntrack-max-per-core=131072 --resource-container=\"\" "
-	cmd += flags
-
-	// cmd += " 1>>/var/log/kube-proxy.log 2>&1"
-	cmd += " 2>&1 | /usr/bin/tee /var/log/kube-proxy.log"
 
-	glog.V(2).Infof("built kube-proxy command: %s", cmd)
-
-	command := []string{
-		"/bin/sh", "-c",
-		cmd,
-	}
+	flags = append(flags, []string{
+		"--conntrack-max-per-core=131072",
+		"--kubeconfig=/var/lib/kube-proxy/kubeconfig",
+		"--oom_score_adj=-998",
```

Comment:
Upstream added the code which did `echo -998 > /proc/$$$/oom_score_adj && ...`, so I opened https://github.com/kubernetes/kubernetes/issues/51083 to ask upstream / @vishh